devito/ir/iet/analysis.py

```diff
@@ -258,13 +262,36 @@ def mark_halospot_hoistable(analysis):
     for i, halo_spots in MapNodes(Iteration, HaloSpot).visit(analysis.iet).items():
         for hs in halo_spots:
             if hs in properties:
-                # Already went through this HaloSpot, let's save some analysis time
+                # Already went through this HaloSpot
                 continue
-            # A sufficient condition to be `hoistable` is that, for a given Function,
-            # there are no anti-dependences in the entire scope.
-            # TODO: This condition can actually be relaxed, by considering smaller
-            # sections of the scope
-            found = [f for f in hs.fmapper if not analysis.scopes[i].d_anti.project(f)]
+
+            found = []
+            scope = analysis.scopes[i]
+            for f, hse in hs.fmapper.items():
+                # The sufficient condition for `f`'s halo-update to be
+                # `hoistable` is that there are no `hs.dimensions`-induced
+                # flow-dependences touching the halo
+                test = True
```

Comment:
wouldn't be safer to set it to `False` and do `try / excepts` below?

Reply:
not sure I get what you mean here?